### PR TITLE
Use GH_TOKEN for release action

### PR DESCRIPTION
**Pull request by sergeyshakhov**

### CODE REVIEW

**Review by kubamarchwicki — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GH_TOKEN }}
```
Comment (hzerip) [inline .github/workflows/release.yml:25]: Hey checkout needs GH token ?

Comment (sergeyshakhov) [reply .github/workflows/release.yml:25]: Not the checkout itself. GH token is needed for the maven release plugin in order to push to the main branch.

**Review by ivan-shevtsov — APPROVED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (hzerip) [reply .github/workflows/release.yml:25]: Then why it's not at release step ?

Comment (pivovarit) [reply .github/workflows/release.yml:25]: I would definitely try @hzerip's option, but I would not be surprized if that GITHUB_TOKEN would be used just during `deploy` which would be useful just for deploying to Github Packages

Comment (hzerip) [reply .github/workflows/release.yml:25]: I'm also not sure why it's needed but if it's needed then it should be at mvn:release step.

Comment (sergeyshakhov) [reply .github/workflows/release.yml:25]: `maven-release-plugin` changes the version in pom.xml file and pushes it to the main branch in git repo. So we need repo to be checked out by the user that has such privileges.

Comment (hzerip) [reply .github/workflows/release.yml:25]: Example usage of mvn release:prepare https://github.com/hazelcast/hazelcast-cloud-common-model/blob/515fa0d088d3c41ad536777e38dcbe955da6ea91/.github/workflows/tag.yml#L38

Comment (pivovarit) [reply .github/workflows/release.yml:25]: @hzerip this is different, we're publishing to GitHub Packages there - I think @sergeyshakhov is right

Comment (hzerip) [reply .github/workflows/release.yml]: Okay sorry for confusion :/

**Pull request merged**